### DISCUSSION TITLE
Implementing React Interaction Hooks in Vanilla JavaScript: Feasibility and Challenges

### DISCUSSION THREAD

**Discussion by totorofly** [floating-ui/floating-ui#2915] [Ideas]
The interactions provided by the React version are fantastic, and I have extensively used the interaction-related hooks from Floating-UI in my React projects. However, some of our projects are based on vanilla JavaScript, so I'm curious if it's possible to replicate the effects of these React interaction hooks using native JS. If it's feasible, what challenges might we face? Or is it not possible at all? Because I think if these interactions could be implemented natively, it would be great to have a layer that could be used across different frameworks, not just limited to React.

**Comment by atomiks** [comment]
They were originally going to be framework-agnostic, however the implementation was incredibly buggy (this was back in early 2022). When I switched to using React natively, it was a lot simpler to build. Now that most of the primitives are built, theoretically they could be "ported" into a framework-agnostic solution more easily than when I was trying to do it from scratch. However, this is quite a bit of work that I currently don't have incentive for, so there are no plans for it at the moment

**Comment by totorofly** [comment]
My team has been developing an editor based on React, but the powerful capabilities of the editor are all based on native JavaScript. Moreover, due to a significant difference between the editor's own rendering mechanism and React's virtual DOM mechanism, there have always been some inherent contradictions in performance when using React. However, our floating solutions are deeply implemented using the React version of Floating-UI. Therefore, I would like to know if there might be an opportunity to provide similar response-related hooks for Solidjs in the future, like those available in the Floating-UI React version. Perhaps in the future, I could try implementing it using Solidjs.
